doc/elements-tx-format.md

```diff
@@ -0,0 +1,570 @@
+
+*Note:* Values are listed in multiples of 100,000,000 (this corresponds directly to how bitcoin encodes value, where 1 BTC is divisible into 100,000,000 satoshis). For example, a `ConfidentialAmount` representing 7 inflation keys would be encoded as 700,000,000.
+
+#### Transaction
+
+|Field|Required|Size|Data Type|Encoding|Notes| 
+|-----|--------|-----|---------|--------|-----|
+| Version | Yes | 4 bytes | `UnsignedInteger` | Little endian | Protocol version (should always be 2). |
+| Flags | Yes | 1 byte | | | Whether the transaction contains a witness. Either 0 or 1. |
+| Num Inputs | Yes | Varies | `VarInt` | | Number of inputs to the transaction. |
+| Inputs | Yes | Varies | `Vector<TxIn>` | | |
+| Num Inputs | Yes | Varies | `VarInt` | | Number of outputs from the transaction. |
+| Outputs | Yes | Varies | `Vector<TxOut>` | | |
+| Locktime | Yes | 4 bytes | `UnsignedInteger` | Little endian | See [BIP 113](https://github.com/bitcoin/bips/blob/master/bip-0113.mediawiki). |
+| Witness | Only if flags is 1 | Varies | `Witness` | | See [BIP 141](https://github.com/bitcoin/bips/blob/master/bip-0141.mediawiki). Note that elements witnesses contain more data than bitcoin witnesses. This extra data is described further below. |
+
+Serialized format: `[Version][Flags][Num Inputs][Inputs][Num Outputs][Outputs][Locktime][Witness]`
+
+This is the overarching structure of a serialized transaction. The rest of this document contains further details on specific parts, as well as examples.
+
+#### TxIn
+
+|Field|Required|Size|Data Type|Encoding|Notes| 
+|-----|--------|-----|---------|--------|-----|
+| Output Hash (TXID) | Yes | 32 bytes | `UnsignedInteger` | Little endian | Hashes in bitcoin are encoded as integers in little endian. Be mindful – this means that they will look backwards when serialized. |
+| Output Index | Yes | 4 bytes | `UnsignedInteger` | Little endian | **Output is a coinbase**: `0xffffffff`<br><br>The two most significant bits are reserved for flags.<br><br>**Output is a pegin:** second most significant bit is 1.<br><br>**Output is an asset issuance:** most significant bit is 1. |
+| ScriptSig Length | Yes | Varies | `VarInt` | | Set to `0x00` if the transaction is segwit and the witness contains the signature. |
+| ScriptSig | If ScriptSig Length is non-zero | Varies | `hex` | Big endian | |
+| Sequence | Yes | 4 bytes | `UnsignedInteger` | Little endian | |
+| Asset Issuance | Only if the transaction has an issuance (as indicated by the Output Index) | Varies | `AssetIssuance` |  | |
+
+Serialized format: `[Output Hash][Output Index][ScriptSig Length][ScriptSig][Sequence][Asset Issuance]`
+
+#### TxOut
+
+|Field|Required|Size|Data Type|Encoding|Notes| 
+|-----|--------|-----|---------|--------|-----|
+| Asset | Yes | 1 or 33 bytes | `ConfidentialAsset` | |  |
+| Amount | Yes | 1 or 9 or 33 bytes | `ConfidentialAmount` | | |
```

Comment:
Null asset and null amounts are illegal in outputs.  They cannot be 1 byte.

Reply:
I've added notes explaining this.